tests/SearchHistoryServiceTest.php

```
@@ -0,0 +1,25 @@
+use App\Service\SearchHistoryService;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+class SearchHistoryServiceTest extends TestCase
+{
+    public function testHistoryIsLimitedAndStored(): void
+    {
+        $session = $this->createMock(SessionInterface::class);
+        $history = [];
+        $session->method('get')->willReturnCallback(fn($key, $default) => $history);
+        $session->expects($this->exactly(5))->method('set')->willReturnCallback(function($key, $val) use (&$history) { $history = $val; });
```

PHPMD (2.15.0)

Avoid unused parameters such as '$key'. (Unused Code Rules)

(UnusedFormalParameter)